cartography/intel/azure/storage.py

```diff
@@ -0,0 +1,800 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+
+import neo4j
+from azure.core.exceptions import ClientAuthenticationError
+from azure.core.exceptions import HttpResponseError
+from azure.core.exceptions import ResourceNotFoundError
+from azure.mgmt.storage import StorageManagementClient
+
+from .util.credentials import Credentials
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_client(credentials: Credentials, subscription_id: str) -> StorageManagementClient:
+    """
+    Getting the Azure Storage client
+    """
+    client = StorageManagementClient(credentials, subscription_id)
+    return client
+
+
+@timeit
+def get_storage_account_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    """
+    Getting the list of storage accounts
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        storage_account_list = list(map(lambda x: x.as_dict(), client.storage_accounts.list()))
+
+    # ClientAuthenticationError and ResourceNotFoundError are subclasses under HttpResponseError
+    except ClientAuthenticationError as e:
+        logger.warning(f"Client Authentication Error while retrieving storage accounts - {e}")
+        return []
+    except ResourceNotFoundError as e:
+        logger.warning(f"Storage Account not found error - {e}")
+        return []
+    except HttpResponseError as e:
```

Comment:
I feel we should raise here. If you get the `ClientAuthenticationError` and `ResourceNotFoundError` exceptions upon not having proper permissions, then that's fine, we should continue the sync, otherwise, the other failure cases could indicate an incomplete sync.